tidy3d/components/medium.py

```diff
@@ -548,208 +1007,142 @@ def eps_diagonal_on_grid(
+        if np.logical_xor(*[isinstance(f, SpatialDataArray) for f in (n, k)]) or np.logical_xor(
+            *[isinstance(f, ScalarFieldDataArray) for f in (n, k)]
+        ):
+            raise SetupError("'n' and 'k' must be of the same type.")
         if n.coords != k.coords:
-            raise SetupError("`n` and `k` must have same coordinates.")
-
-        eps_values = Medium.nk_to_eps_complex(n=n.data, k=k.data)
-        coords = {k: np.array(v) for k, v in n.coords.items()}
-        eps_scalar_field_data = ScalarFieldDataArray(eps_values, coords=coords)
-        return cls.from_eps_raw(eps=eps_scalar_field_data, interp_method=interp_method)
-
-    @staticmethod
-    def _interp(
-        scalar_dataset: ScalarFieldDataArray,
-        coord_interp: Coords,
-        interp_method: InterpMethod,
-    ) -> ScalarFieldDataArray:
-        """
-        Enhance xarray's ``.interp`` in two ways:
-            1) Check if the coordinate of the supplied data are in monotically increasing order.
-            If they are, apply the faster ``assume_sorted=True``.
-
-            2) For axes of single entry, instead of error, apply ``isel()`` along the axis.
-
-            3) When linear interp is applied, in the extrapolated region, filter values smaller
-            or larger than the original data's min(max) will be replaced with the original min(max).
+            raise SetupError("'n' and 'k' must have same coordinates.")
 
-        Parameters
-        ----------
-        scalar_dataset : :class:`.ScalarFieldDataArray`
-            Supplied scalar dataset.
-        coord_interp : :class:`.Coords`
-            The grid point coordinates over which interpolation is performed.
-        interp_method : :class:`.InterpMethod`
-            Interpolation method.
-
-        Returns
-        -------
-        :class:`.ScalarFieldDataArray`
-            The interpolated scalar dataset.
-        """
-
-        # check in x/y/z axes, which of them are supplied with a single entry.
-        all_coords = "xyz"
-        is_single_entry = [scalar_dataset.sizes[ax] == 1 for ax in all_coords]
-        interp_ax = [
-            ax for (ax, single_entry) in zip(all_coords, is_single_entry) if not single_entry
-        ]
-        isel_ax = [ax for ax in all_coords if ax not in interp_ax]
-
-        # apply isel for the axis containing single entry
-        if len(isel_ax) > 0:
-            scalar_dataset = scalar_dataset.isel(
-                {ax: [0] * len(coord_interp.to_dict[ax]) for ax in isel_ax}
-            )
-            scalar_dataset = scalar_dataset.assign_coords(
-                {ax: coord_interp.to_dict[ax] for ax in isel_ax}
+        # k is a SpatialDataArray
+        if isinstance(k, SpatialDataArray):
+            if freq is None:
+                raise SetupError(
+                    "For a lossy medium, 'freq' at which 'eps' is defined must be supplied",
+                )
+            eps_inf, sigma = CustomMedium.nk_to_eps_sigma(n, k, freq)
```

Comment:
minor point for clarity, but this probably isn't "eps_inf" right? just eps evaluated at f=freq?

Reply:
sure sounds good